cli/internal/codegen/testdata/expected_golang.go

```diff
@@ -84,11 +89,87 @@ func WithAuthFunc(tokenGenerator func(ctx context.Context) (string, error)) Opti
 	}
 }
 
+type AuthenticationUser struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type ProductsCreateProductRequest struct {
+	IdempotencyKey string `header:"Idempotency-Key"`
+	Name           string `json:"name"`
+	Description    string `json:"description,omitempty"`
+}
+
+type ProductsProduct struct {
+	ID          string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description,omitempty"`
+	CreatedAt   time.Time          `json:"created_at"`
+	CreatedBy   AuthenticationUser `json:"created_by"`
+}
+
+type ProductsProductListing struct {
+	Products     []ProductsProduct `json:"products"`
+	PreviousPage struct {
+		Cursor string `json:"cursor,omitempty"`
+		Exists bool   `json:"exists"`
+	} `json:"previous"`
+	NextPage struct {
+		Cursor string `json:"cursor,omitempty"`
+		Exists bool   `json:"exists"`
+	} `json:"next"`
+}
+
+// ProductsClient Provides you access to call public and authenticated APIs on products. The concrete implementation is productsClient.
+// It is setup as an interface allowing you to use GoMock to create mock implementations during tests.
+type ProductsClient interface {
+	Create(ctx context.Context, params ProductsCreateProductRequest) (ProductsProduct, error)
+	List(ctx context.Context) (ProductsProductListing, error)
+}
+
+type productsClient struct {
+	base *baseClient
+}
+
+var _ ProductsClient = (*productsClient)(nil)
+
+func (c *productsClient) Create(ctx context.Context, params ProductsCreateProductRequest) (resp ProductsProduct, err error) {
+	// Convert our params into the objects we need for the request
+	headers := map[string][]string{"Idempotency-Key": {params.IdempotencyKey}}
```

Comment:
Should we maybe use [http.Header](https://pkg.go.dev/net/http#Header) here? Would look a bit more standard.

Reply:
I went with a basic map over a `http.Headers` because via the headers API you can't get a list of the keys - but you can cast it to a map. I guess I was thinking if the headers type gets changed in the future while keeping the API consistent we'd need to make a breaking change to swap to a type where we could read all the keys.

That said I'll change it, because again I think this is a minimal risk

Reply:
`http.Headers` is exactly the same data structure under the hood (`map[string][]string`) so I figured it would behave identically?

Reply:
It is, but it's a named type with it's own API. So I think it's safe to change - but my fear was what if they changed the named type to a struct with the map in it for some reason.

As I said, changing the code to reference `http.Header` 😄

Reply:
Ah I understand, yeah they cannot change that since iterating over the map is part of the public API.